flake8_annotations/__init__.py

```diff
@@ -157,16 +174,22 @@ def get_class_decorator_type(
         If @classmethod or @staticmethod decorators are not present, this function will return None
         """
         decorators = []
+        decorator_attributes = []
         for decorator in function_node.decorator_list:
-            # @classmethod and @staticmethod will show up as ast.Name objects, where callable
-            # decorators will show up as ast.Call, which we can ignore
+            # @property, @classmethod, and @staticmethod will be ast.Name objects
+            # property.setter, and property.deleter will be ast.Attribute objects
+            # Other callable decorators will be ast.Call objects, which we're ignoring
             if isinstance(decorator, ast.Name):
                 decorators.append(decorator.id)
+            elif isinstance(decorator, ast.Attribute):
+                decorator_attributes.append(decorator.attr)
```

Comment:
Will this only catch `property.setter` and `property.deleter` or will it catch attributes of _any_ decorator?

Reply:
Any decorator. I don't know if it's a good one, but the assumption being made is that anything with these attributes (`.getter`, `.setter`, `.deleter`) is a property method.

I can't think of a better alternative short of refactoring the entire AST parser to collect class methods together, by class, and parsing the list of functions to see if the decorators with this subset of attributes correspond to a property of the class. This seems like a lot of work for little gain, nor would be be able to well handle subclasses that may overload say, a setter but not the getter. At this point we're departing what we can adequately handle from a static analysis perspective without significant effort (or even at all) and entering into the domain of something like mypy.

I'm still not wedded to the idea of separating out property decorators in our linting, especially less so if we're increasing our chances of mischaracterization.

Reply:
If separating decorators was actually useful then I'd say it's fine to make that assumption. But given this isn't the case, I am now inclined to just remove this functionality if it can't work perfectly.

Reply:
I've created a [branch](https://github.com/python-discord/flake8-annotations/tree/non-property-fixes) with the non-property related updates should we decide to scrap the property methods for the above reasons (and those discussed in #20).

Shower us with opinions pls :)

Reply:
I guess I agree with Mark. property decorators were nice when they were free but at this first sign of trouble I'm absolutely fine just ditching them.